src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs

```diff
@@ -160,20 +173,47 @@ public IMoBiCommand RemoveFormulaUsablePath(IFormula formula, FormulaUsablePath
       {
-         var command = new EditPathAtUsablePathCommand(formula, newPath, formulaUsablePath, buildingBlock).RunCommand(_context);
-         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
-      }
+         IMoBiCommand command;
+         if (isTimePathAndDimension(newPath, formulaUsablePath.Dimension))
+         {
+            var timePath = _context.ObjectPathFactory.CreateTimePath(timeDimension).WithAlias(formulaUsablePath.Alias);
+            command = swapFormulaUsablePaths(formula, formulaUsablePath, timePath, buildingBlock);
+         }
+         else if (formulaUsablePath is TimePath)
+         {
+            var newFormulaUsablePath = _context.ObjectPathFactory.CreateFormulaUsablePathFrom(newPath).WithAlias(formulaUsablePath.Alias).WithDimension(formulaUsablePath.Dimension);
+            command = swapFormulaUsablePaths(formula, formulaUsablePath, newFormulaUsablePath, buildingBlock);
+         }
+         else
+            command = new EditPathAtUsablePathCommand(formula, newPath, formulaUsablePath, buildingBlock);
 
-      public IMoBiCommand AddFormulaUsablePath(IFormula formula, FormulaUsablePath path, IBuildingBlock buildingBlock)
-      {
-         return new AddFormulaUsablePathCommand(formula, path, buildingBlock).RunCommand(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command.RunCommand(_context), formula, buildingBlock);
       }
 
-      public IMoBiCommand ChangeVariableName(SumFormula formula, string newVariableName, IBuildingBlock buildingBlock)
+      private IMoBiCommand swapFormulaUsablePaths(IFormula formula, FormulaUsablePath oldPath, FormulaUsablePath newPath, IBuildingBlock buildingBlock)
       {
-         return new ChangeVariableNameCommand(formula, newVariableName, buildingBlock).RunCommand(_context);
+         var macroCommand = new MoBiMacroCommand
+         {
+            Description = AppConstants.Commands.ChangeFormulaUsablePath,
+            ObjectType = ObjectTypes.Formula,
+            CommandType = AppConstants.Commands.EditCommand
+         };
+
+         macroCommand.Add(getRemoveFormulaUsablePathCommand(formula, oldPath, buildingBlock));
+         macroCommand.Add(getAddFormulaUsablePathCommand(formula, newPath, buildingBlock));
+         return macroCommand;
       }
 
+      private bool isTimePathAndDimension(ObjectPath newPath, IDimension dimension) => dimension.Equals(timeDimension) && newPath.PathAsString.Equals(Constants.TIME);
+
+      public IMoBiCommand AddFormulaUsablePath(IFormula formula, FormulaUsablePath path, IBuildingBlock buildingBlock) => getAddFormulaUsablePathCommand(formula, path, buildingBlock).RunCommand(_context);
+
+      private static AddFormulaUsablePathCommand getAddFormulaUsablePathCommand(IFormula formula, FormulaUsablePath path, IBuildingBlock buildingBlock) => new AddFormulaUsablePathCommand(formula, path, buildingBlock);
```

Comment:
why these methods? they are just doing new with the parameter ? looking at the code, I thoight something big was happening and in fact it just creates an instance. Doyou think it helps readability?

Reply:
Yeah, could do without it. It didn't really help with reuse since the only thing was creating the command